Lib/test/test_bool.py

```diff
@@ -58,8 +58,10 @@ def test_math(self):
         self.assertEqual(-True, -1)
         self.assertEqual(abs(True), 1)
         self.assertIsNot(abs(True), True)
-        self.assertEqual(~False, -1)
-        self.assertEqual(~True, -2)
+        with self.assertWarns(DeprecationWarning):
+            self.assertEqual(~False, -1)
```

Comment:
The deprecation warning is not issued during testing - and thus the test suite fails.

Can anybody give a hint on why that's the case? When compiling locally, I get the warning shown in the interpreter.

Reply:
How I understand, this is a some sort of optimization(?).
Just run this code:
```python
def foo():
    return ~False
```
As you see, there's no call of function `foo`. But you'll see a `DeprecationWarning`.
I doesn't completely understand why is this so, but, simplest way to force this test work looks like that:
```python
with self.assertWarns(DeprecationWarning):
    eval("~False")
```

Reply:
Yeah, the reason is constant folding in the compiler:
```
In [1]: import dis

In [2]: def foo():
   ...:     x = 1 + 2 + 3
   ...:     y = ~False
   ...: 

In [3]: dis.dis(foo)
  2           0 LOAD_CONST               1 (6)
              2 STORE_FAST               0 (x)

  3           4 LOAD_CONST               2 (-1)
              6 STORE_FAST               1 (y)
              8 LOAD_CONST               0 (None)
             10 RETURN_VALUE
```

Reply:
Thanks for the clarification.

I've switched from
```
eval("~False")
```
to
``` 
false = False
~false
```
which also prevents the constant folding and is hopefully less obscure than `eval()`